### PR TITLE
class2-rockpaperscissors

### PR DESCRIPTION
**Pull request by mlukso**
## Checkpoint Rubric
This is the rubric that your instructor will use to grade your checkpoints. Please do not edit. 

**Checkpoint 1**
- [ ] All tests passed: 40 points
- [ ] Proper use of documentation (commenting on code): 15 points
- [ ] Properly indented code: 15 points
- [ ] Demonstrated effective use of JavaScript: 30 points

**Checkpoint 2**
- [ ] The application works as it should: 40 points
- [ ] Proper use of documentation (commenting on code): 15 points
- [ ] Properly indented code: 15 points
- [ ] Demonstrated effective use of JavaScript and the DOM API: 30 points

**Checkpoint 3**
- [ ] Use of React: 25 points
- [ ] Accesses an API: 25 points
- [ ] Proper use of documentation (commenting on code): 25 points
- [ ] The application functions as it should: 25 points

### CODE REVIEW

File: 01week/rockPaperScissors.js [hunk 1]
```diff
@@ -7,31 +7,40 @@ const rl = readline.createInterface({
...
 
+
+const rockPaperScissors=(hand1, hand2)=>{
+
+  // Check if inputs are "Truthy"
+  if (hand1 && hand2){
+  } else {return "Please enter 'rock', 'paper', or 'scissors'"}
+
+  // Trim spaces, convert to lowercase, and ensure a String for each of the inputs
+  hand1 = hand1.trim();
```
Comment (reneemeyer) [inline 01week/rockPaperScissors.js]: In functional programming we never want to change our original parameters. Best practice would be to make a new variable and assign to the the trimmed variable. The last string method is unnecessary and redundant 
```javascript
const hand1Formatted = hand1.trim().toLowerCase();
```

File: 01week/rockPaperScissors.js [hunk 2]
```diff
@@ -7,31 +7,40 @@ const rl = readline.createInterface({
...
+  hand1 = hand1.toLowerCase();
+  hand1 = String(hand1)
+
+  hand2 = hand2.trim();
+  hand2 = hand2.toLowerCase();
+  hand2 = String(hand2)
+
+  // game logic to compare inputs
+  if (hand1 !== 'rock' && hand1 !== 'paper' && hand1 !== 'scissors'){return "Please enter 'rock', 'paper', or 'scissors'"}
```
Comment (reneemeyer) [inline 01week/rockPaperScissors.js]: This should be formatted for better readability. Return on a new line and indented, curly brackets should line up.

File: 01week/rockPaperScissors.js [hunk 3]
```diff
@@ -7,31 +7,40 @@ const rl = readline.createInterface({
...
+  else if (hand2 !== 'rock' && hand2 !== 'paper' && hand2 !== 'scissors'){return "Please enter 'rock', 'paper', or 'scissors'"}
+
+    else if (hand1 === hand2){return "It's a tie!"}
```
Comment (reneemeyer) [inline 01week/rockPaperScissors.js]: This should be formatted for better readability. Return on a new line and indented, curly brackets should line up.

File: 01week/rockPaperScissors.js [hunk 4]
```diff
@@ -7,31 +7,40 @@ const rl = readline.createInterface({
...
+    else if (hand1 === 'paper' && hand2 === 'rock') {return "Hand one wins!"}
```
Comment (reneemeyer) [inline 01week/rockPaperScissors.js]: This should be formatted for better readability. Return on a new line and indented, curly brackets should line up.

**Review by reneemeyer — CHANGES_REQUESTED** [review]
Please see comments below

File: 01week/rockPaperScissors.js [hunk 5]
```diff
@@ -7,17 +7,50 @@ const rl = readline.createInterface({
...
+  // else return "Hand two wins!"
 
-function rockPaperScissors(hand1, hand2) {
 
-  // Write code here
+const rockPaperScissors=(hand1, hand2)=>{
 
+  // Check if inputs are "Truthy"
+  if (hand1 && hand2){
+  } else {
```
Comment (reneemeyer) [inline 01week/rockPaperScissors.js:24]: this empty if statement doesn't read well, I would do 
```javascript
if(!hand1 || ! hand2) return ....
```

**Review by reneemeyer — COMMENTED** [review]
nice job, merge it in

File: 01week/rockPaperScissors.js (hunk 5 above)
Comment (mlukso) [reply 01week/rockPaperScissors.js:24]: Ah, good catch. Thanks!

**Pull request merged**